x-pack/plugins/apm/public/context/environments_context/environments_context.tsx

```diff
@@ -36,8 +41,14 @@ export function EnvironmentsContextProvider({
     ('environment' in query && (query.environment as Environment)) ||
     ENVIRONMENT_ALL.value;
 
-  const rangeFrom = 'rangeFrom' in query ? query.rangeFrom : undefined;
-  const rangeTo = 'rangeTo' in query ? query.rangeTo : undefined;
+  /* eslint-disable prettier/prettier */
+  const rangeFrom = customTimeRange
+    ? customTimeRange.rangeFrom
+    : 'rangeFrom' in query ? query.rangeFrom : undefined;
+  const rangeTo = customTimeRange
+    ? customTimeRange.rangeTo
+    : 'rangeTo' in query ? query.rangeTo : undefined;
```

Comment:
Maybe this will look a bit cleaner?!

```suggestion
const queryRangeFrom = 'rangeFrom' in query  ? query.rangeFrom : undefined;
const queryRangeTo = 'rangeTo' in query  ? query.rangeTo : undefined;

  const rangeFrom = customTimeRange?.rangeFrom || queryRangeFrom
  const rangeTo = customTimeRange?.rangeTo || queryRangeTo

```